server/authmanager.go

```diff
@@ -108,18 +106,30 @@ func (a *JWTAuthManager) stop() {
 func (a *JWTAuthManager) replaceJWT(privateKey *rsa.PrivateKey, kid string, jwtExpiration time.Duration) error {
 	log.Debugf("setting internal JWT")
 
-	token, err := NewToken(jwtExpiration)
+	rsaSigner, err := jose.NewSigner(
```

Comment:
Also, seems like each auth manager is only handling one set of private key and kid, what do you think of a refactoring that reuses this signer?

Reply:
I think it's unnecessary, there is only one AuthManager.

Reply:
Sure. This is just out of scope of this work, so if we want to track, I think another ticket would be best - but also, this will likely be replaced when we have a formal API instead of the proxy anyway.